fastplotlib/graphics/selectors/_rectangle.py

```diff
@@ -199,6 +199,43 @@ def __init__(
             edge.world.z = -0.5
             group.add(edge)
 
+        # vertices
+        top_left_vertex_data = (xmin, ymax, 1)
+        top_right_vertex_data = (xmax, ymax, 1)
+        bottom_left_vertex_data = (xmin, ymin, 1)
+        bottom_right_vertex_data = (xmax, ymin, 1)
+
+        top_left_vertex = pygfx.Points(
+            pygfx.Geometry(positions=[top_left_vertex_data], sizes=[edge_thickness]),
+            pygfx.PointsMaterial(size=edge_thickness, color=edge_color, size_space="world", size_mode="vertex"),
+        )
+
+        top_right_vertex = pygfx.Points(
+            pygfx.Geometry(positions=[top_right_vertex_data], sizes=[edge_thickness]),
+            pygfx.PointsMaterial(size=edge_thickness, color=edge_color, size_space="world", size_mode="vertex"),
+        )
+
+        bottom_left_vertex = pygfx.Points(
+            pygfx.Geometry(positions=[bottom_left_vertex_data], sizes=[edge_thickness]),
+            pygfx.PointsMaterial(size=edge_thickness, color=edge_color, size_space="world", size_mode="vertex"),
+        )
+
+        bottom_right_vertex = pygfx.Points(
+            pygfx.Geometry(positions=[bottom_right_vertex_data], sizes=[edge_thickness]),
+            pygfx.PointsMaterial(size=edge_thickness, color=edge_color, size_space="world", size_mode="vertex"),
+        )
+
+        self.vertices: Tuple[pygfx.Points, pygfx.Points, pygfx.Points, pygfx.Points] = (
+            bottom_left_vertex,
+            bottom_right_vertex,
+            top_left_vertex,
+            top_right_vertex,
+        )
+
+        for vertex in self.vertices:
+            vertex.world.z = -0.25
```

Comment:
why -0.25 :thinking:

Reply:
the edges get set to -0.5, so if the vertices are -0.25 then they will be on top of the edges

Reply:
what do you mean?

Reply:
Sometimes world objects will have weird overlapping when the z-planes are close with very high or very low zoom, I just tried it and it seems fine within reasonable levels.